modules/tasks/taskClaim.js

```diff
@@ -22,38 +25,43 @@ const sendConfirmationEmail = (task, user, comments) => {
   )
 }
 
-const verifyIssueAndClaim = async (task, user, comments) => {
+const verifyIssueAndClaim = async (task, user, comments, token) => {
   const language = user.language || 'en'
   i18n.setLocale(language)
 
-  if (task.provider === 'github' && user.provider === task.provider) {
-    if (task.metadata.user === user.provider_username) {
-      const taskUser = task.user
-
-      // Update task
-      await taskUpdate({ id: task.id, userId: user.id })
-
-      const body = `${i18n.__('mail.task.claim.confirmation.body', {
-        title: task.title,
-        url: task.url,
-        comments: comments
-      })}`
+  return jwt.verify(token, process.env.SECRET_PHRASE, async (err, decoded) => {
+    // the 401 code is for unauthorized status
+    if (err || decoded !== task.id) {
+      throw new Error('invalid_token')
+    }
 
-      return SendMail.success(
-        { email: taskUser.email, language },
-        i18n.__('mail.task.claim.confirmation.subject'), body
-      )
+    if (task.provider !== 'github' || user.provider !== task.provider) {
+      throw new Error('invalid_provider')
     }
-    else {
+
+    if (task.metadata.user !== user.provider_username) {
```

Comment:
```suggestion
    if (task.metadata.issue.user.login !== user.provider_username) {
```

In my case, the `task.metadata.user` it was the organization, so that's why its not working. I suggest checking the issue author instead.